src/utils/mlflow_utils.py

```diff
@@ -10,76 +13,280 @@
 
 
-def check_mlflow_ready():
-    """
-    Check if MLflow is reachable.
-
-    Raises:
-        Exception: If MLflow is not reachable
-    """
-    try:
-        os.environ['MLFLOW_TRACKING_USERNAME'] = MLFLOW_TRACKING_USERNAME
-        os.environ['MLFLOW_TRACKING_PASSWORD'] = MLFLOW_TRACKING_PASSWORD
-        mlflow.set_tracking_uri(MLFLOW_TRACKING_URI)
-        client = MlflowClient()  # noqa: F841
-        return True
-    except Exception as e:
-        logger.warning(f"MLflow is not reachable: {e}")
-        return False
-
-
-def get_mlflow_models():
-    """
-    Retrieve available MLflow models and create dropdown options.
-    Models are sorted by creation timestamp, from newest to oldest.
-
-    Returns:
-        list: Dropdown options for MLflow models
-    """
-    try:
-        # MLflow configuration
-        os.environ['MLFLOW_TRACKING_USERNAME'] = MLFLOW_TRACKING_USERNAME
-        os.environ['MLFLOW_TRACKING_PASSWORD'] = MLFLOW_TRACKING_PASSWORD
-        mlflow.set_tracking_uri(MLFLOW_TRACKING_URI)
-
-        client = MlflowClient()
-
-        # Search for registered models
-        registered_models = client.search_registered_models()
-
-        # Sort models by creation timestamp (newest first)
-        registered_models = sorted(
-            registered_models, key=lambda model: model.creation_timestamp, reverse=True
+class MLflowClient:
+    """A wrapper class for MLflow client operations."""
+    
+    # In-memory model cache (for quick access)
+    _model_cache = {}
+    
+    def __init__(
+        self, 
+        tracking_uri=None,
+        username=None, 
+        password=None,
+        cache_dir=None
+    ):
+        """
+        Initialize the MLflow client with connection parameters.
+        
+        Args:
+            tracking_uri: MLflow tracking server URI
+            username: MLflow authentication username
+            password: MLflow authentication password
+            cache_dir: Directory to store cached models
+        """
+        self.tracking_uri = tracking_uri or os.getenv("MLFLOW_TRACKING_URI")
+        self.username = username or os.getenv("MLFLOW_TRACKING_USERNAME", "")
+        self.password = password or os.getenv("MLFLOW_TRACKING_PASSWORD", "")
+        self.cache_dir = cache_dir or MLFLOW_CACHE_DIR
+        
+        # Create cache directory if it doesn't exist
+        os.makedirs(self.cache_dir, exist_ok=True)
+        
+        # Set environment variables
+        os.environ['MLFLOW_TRACKING_USERNAME'] = self.username
+        os.environ['MLFLOW_TRACKING_PASSWORD'] = self.password
+        
+        # Set tracking URI
+        mlflow.set_tracking_uri(self.tracking_uri)
+        
+        # Create client
+        self.client = MlflowClient()
+
+    def check_mlflow_ready(self):
+        """
+        Check if MLflow is reachable.
+
+        Raises:
+            Exception: If MLflow is not reachable
+        """
+        try:
+            # Use the existing client
+            _ = self.client
```

Comment:
I do not think this would trigger an error if the server is not running. Perhaps using something like `client.search_experiments(max_results=1)` can help - but I noticed it took a couple of seconds to return an error at my end.